packages/oauth/src/index.ts

```diff
@@ -7,11 +7,29 @@ export interface OAuthProvider {
 		providerUser: Record<string, any>;
 		[data: string]: any;
 	}>;
-	getAuthorizationUrl: () => string;
+	getAuthorizationUrl: (state?: string) => [url: string, state: string | undefined];
```

Comment:
So the issue here is the return type. The returned `state` should be undefined if `OAuthConfig.state` is `false` but I think it’s impossible since the config is optional (it’s easy if the type is only `boolean` and not `boolean | undefined`).

The only solution that makes sense that works with TypeScript is to make the optional argument for `getAuthorizationUrl()` have a type of `string | null | undefined`, where the state is *not* included if `null`.

Reply:
If you can make the return type change based on the value/type of `OAuthConfig.state`, I’d like to know!

Reply:
> The only solution that makes sense that works with TypeScript is to make the optional argument for `getAuthorizationUrl()` have a type of `string | null | undefined`, where the state is _not_ included if `null`.

Ah, I see what you mean. I like that. It'll make it much more straightforward in how the `state` is handled when it's entirely configured in `getAuthorizedUrl()`.

Reply:
The only thing I'm not 100% sure about is how to handle the case where an empty string is passed in. In the newest iteration, the `state` won't be appended to the auth url, but it does return the empty string back in the tuple.

Reply:
I think it does append the value and the url search params will be `?state=`, which is expected.